src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -122,14 +128,12 @@ class GenesisDataLoader(
       case Some(existingGenesisHeader) if existingGenesisHeader.hash == header.hash =>
         log.info("Genesis data already in the database")
         Success(())
-
       case Some(_) =>
         Failure(new RuntimeException("Genesis data present in the database does not match genesis block from file." +
           " Use different directory for running private blockchains."))
-
       case None =>
         // using empty namespace because ephemDataSource.storage already has the namespace-prefixed keys
-        dataSource.update(IndexedSeq(), Nil, ephemDataSource.storage.toSeq)
+        ephemDataSource.storage.toSeq.grouped(dbConfig.batchSize).foreach(toStore => dataSource.update(IndexedSeq(), Nil, toStore))
```

Comment:
@ntallar I do agree with you but I wanted to avoid hiding it underneath `io.iohk.ethereum.db.dataSource.LevelDBDataSource#update`. It would be misleading because instead of doing an atomic update we would be creating several batches to avoid this issue. But if it's better, I can change it tho. @mcsherrylabs @LukasGasior1 what do you think?